examples/compiler/iet/tut_1.ipynb

```diff
@@ -0,0 +1,1162 @@
+    "u = TimeFunction(name='u', grid=grid)\n",
+    "u.data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here, we have just declared a two-dimensional domain with three coordinates at each dimension ($x$ and $y$). Each coordinate of such a discrete space will be holding a $real~value$.\n",
+    "\n",
+    "As we can see, we can always access the values at each coordinate of the domain. At this point, no modifications have been done to it yet. `u.data` give as a quick access over the values holded by each cell of such a domain. `u.data[0]` holds the values in the grid at the \"current\" iteration time, given a time-step to be considered, whereas `u.data[1]` holds the values of `u` for the \"current+1\" time-step.\n",
+    "\n",
+    "We can now create an `operator` that will perform modifications onto our domain according to $differential~ equations$ through a computational stencil. \n",
+    "It means that those differential equations will be translated into finite differences that will be used to update the values at each coordinate of the space.\n",
+    "Such finite differences, or `expressions`, will be applied for specific ranges of `iterations` over the domain. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Eq(u(t + dt, x, y), u(t, x, y) + 1)"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "eq = Eq(u.forward, u+1)\n",
+    "op = Operator(eq)\n",
+    "op.args['expressions']"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For instance, the particular `equation` object above allows us say that, at each time step, `1` will be added to every position of the domain.\n",
+    "\n",
+    "Let's take a look at the $kernel$ that will be used to compute how this equation alters the domain."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "#define _POSIX_C_SOURCE 200809L\n",
+      "#include \"stdlib.h\"\n",
+      "#include \"math.h\"\n",
+      "#include \"sys/time.h\"\n",
+      "#include \"xmmintrin.h\"\n",
+      "#include \"pmmintrin.h\"\n",
+      "\n",
+      "struct Profiler\n",
+      "{\n",
+      "  double section0;\n",
+      "} ;\n",
+      "\n",
+      "\n",
+      "int Kernel(float *restrict u_vec, const int time_M, const int time_m, void *_timers, const int x_M, const int x_m, const int x_size, const int y_M, const int y_m, const int y_size)\n",
+      "{\n",
+      "  float (*restrict u)[x_size + 1 + 1][y_size + 1 + 1] __attribute__((aligned(64))) = (float (*)[x_size + 1 + 1][y_size + 1 + 1]) u_vec;\n",
+      "  struct Profiler *timers = (struct Profiler*) _timers;\n",
```

Comment:
These type casts will be dropped by #620. Could you move this tutorial on top of #620 and refresh the various cells so that they are up-to-date?